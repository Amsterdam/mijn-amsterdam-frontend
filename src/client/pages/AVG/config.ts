import { AVGRequestFrontend } from '../../../server/services/avg/types';
import { dateSort } from '../../../universal/helpers/date';
import { LinkProps } from '../../../universal/types';
import {
  DisplayProps,
  WithDetailLinkComponent,
} from '../../components/Table/TableV2';

export const displayPropsAanvragen: DisplayProps<
  WithDetailLinkComponent<AVGRequestFrontend>
> = {
  detailLinkComponent: 'Nummer',
  ontvangstDatumFormatted: 'Ontvangen op',
  themas: 'Onderwerp(en)',
};

export const listPageParamKind = {
  inProgress: 'lopende-aanvragen',
  completed: 'afgehandelde-aanvragen',
};

const tableConfigBase = {
  sort: dateSort('dateRequest', 'desc'),
  displayProps: displayPropsAanvragen,
};

export const tableConfig = {
  [listPageParamKind.inProgress]: {
    title: 'Lopende aanvragen',
    filter: (avgVerzoek: AVGRequestFrontend) => !avgVerzoek.datumAfhandeling,
    ...tableConfigBase,
  },
  [listPageParamKind.completed]: {
    title: 'Afgehandelde aanvragen',
    filter: (avgVerzoek: AVGRequestFrontend) => avgVerzoek.datumAfhandeling,
    ...tableConfigBase,
  },
};

export const linkListItems: LinkProps[] = [
  {
    to: 'https://www.amsterdam.nl/privacy/loket/',
    title: 'Loket persoonsgegevens gemeente Amsterdam',
  },
];
