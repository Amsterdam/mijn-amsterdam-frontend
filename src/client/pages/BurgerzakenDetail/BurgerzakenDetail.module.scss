@import '../../styles/_global.scss';

.BurgerzakenDetail {

}

