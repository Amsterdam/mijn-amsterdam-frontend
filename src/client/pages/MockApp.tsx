import { RecoilRoot, RecoilRootProps } from 'recoil';
import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';

interface MockAppProps {
  routePath: string;
  routeEntry: string;
  initializeState?: RecoilRootProps['initializeState'];
  component: React.Component | any;
}

export default function MockApp({
  routePath,
  routeEntry,
  initializeState,
  component,
}: MockAppProps) {
  return (
    <RecoilRoot initializeState={initializeState}>
      <MemoryRouter initialEntries={[routeEntry]}>
        <Route path={routePath} component={component} />
      </MemoryRouter>
    </RecoilRoot>
  );
}
