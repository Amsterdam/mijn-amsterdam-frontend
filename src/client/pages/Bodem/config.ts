import { generatePath } from 'react-router-dom';

import { LoodMetingFrontend } from '../../../server/services/bodem/types';
import { AppRoutes } from '../../../universal/config/routes';
import { dateSort } from '../../../universal/helpers/date';
import { LinkProps } from '../../../universal/types';

const displayPropsAanvragen = {
  detailLinkComponent: 'Adres',
  datumAanvraagFormatted: 'Aangevraagd',
  status: 'Status',
};

const listPageParamKind = {
  inProgress: 'lopende-aanvragen',
  completed: 'afgehandelde-aanvragen',
};

const tableConfigBase = {
  sort: dateSort('dateRequest', 'desc'),
  displayProps: displayPropsAanvragen,
};

export const tableConfig = {
  [listPageParamKind.inProgress]: {
    ...tableConfigBase,
    title: 'Lopende aanvragen',
    filter: (bodemAanvraag: LoodMetingFrontend) => !bodemAanvraag.processed,
    listPageRoute: generatePath(AppRoutes['BODEM/LIST'], {
      kind: listPageParamKind.inProgress,
    }),
  },
  [listPageParamKind.completed]: {
    ...tableConfigBase,
    title: 'Afgehandelde aanvragen',
    filter: (bodemAanvraag: LoodMetingFrontend) => bodemAanvraag.processed,
    listPageRoute: generatePath(AppRoutes['BODEM/LIST'], {
      kind: listPageParamKind.completed,
    }),
  },
} as const;

export const linkListItems: LinkProps[] = [
  {
    title: 'Meer informatie over lood in de bodem.',
    to: 'https://www.amsterdam.nl/wonen-leefomgeving/bodem/lood-grond/',
  },
] as const;
