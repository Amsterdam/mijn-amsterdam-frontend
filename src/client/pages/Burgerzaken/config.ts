import { generatePath } from 'react-router-dom';

import { AppRoutes } from '../../../universal/config/routes';
import { dateSort } from '../../../universal/helpers/date';
import { IdentiteitsbewijsFrontend, LinkProps } from '../../../universal/types';
import { withOmitDisplayPropsForSmallScreens } from '../../components/Table/helpers';
import {
  DisplayProps,
  WithDetailLinkComponent,
} from '../../components/Table/TableV2';

const displayPropsBase: DisplayProps<
  WithDetailLinkComponent<IdentiteitsbewijsFrontend>
> = {
  detailLinkComponent: 'Type',
  datumAfloopFormatted: 'Geldig tot',
};

const listPageParamKind = {
  identiteitsbewijzen: 'identiteitsbewijzen',
};

const displayProps = withOmitDisplayPropsForSmallScreens(displayPropsBase, [
  'datumAfloopFormatted',
]);

export const tableConfig = {
  [listPageParamKind.identiteitsbewijzen]: {
    title: 'Mijn reisdocumenten',
    displayProps,
    sort: dateSort('datumAfloop', 'desc'),
    listPageRoute: generatePath(AppRoutes['BURGERZAKEN/LIST'], {
      kind: listPageParamKind.identiteitsbewijzen,
    }),
  },
} as const;

export const linkListItems: LinkProps[] = [
  {
    to: 'https://www.amsterdam.nl/burgerzaken',
    title: 'Overzicht en aanvragen bij burgerzaken',
  },
] as const;

export const backLinkDetailPage = AppRoutes.BURGERZAKEN;
