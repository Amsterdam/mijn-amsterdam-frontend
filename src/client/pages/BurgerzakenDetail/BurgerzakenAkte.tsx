import { useParams } from 'react-router-dom';
import { AppRoutes, ChapterTitles } from '../../../universal/config';
import {
  capitalizeFirstLetter,
  isError,
  isLoading,
} from '../../../universal/helpers';
import {
  Alert,
  ChapterIcon,
  DetailPage,
  InfoDetail,
  LoadingContent,
  PageContent,
  PageHeading,
} from '../../components';
import DocumentList from '../../components/DocumentList/DocumentList';
import { useAppStateGetter } from '../../hooks/useAppState';
import styles from './BurgerzakenDetail.module.scss';

export default function BurgerzakenIDKaart() {
  const { AKTES } = useAppStateGetter();
  const { id } = useParams<{ id: string }>();

  const Akte = AKTES.content?.find((item) => item.id === id);
  const noContent = !isLoading && !Akte;

  return (
    <DetailPage>
      <PageHeading
        icon={<ChapterIcon />}
        backLink={{
          to: AppRoutes.BURGERZAKEN,
          title: ChapterTitles.BURGERZAKEN,
        }}
        isLoading={isLoading(AKTES)}
      >
        {capitalizeFirstLetter(Akte?.type || 'Akte')}
      </PageHeading>

      <PageContent className={styles.DetailPageContent}>
        {(isError(AKTES) || noContent) && (
          <Alert type="warning">
            <p>We kunnen op dit moment geen gegevens tonen.</p>
          </Alert>
        )}
        {isLoading(AKTES) && (
          <LoadingContent className={styles.LoadingContentInfo} />
        )}
        {!!Akte && (
          <>
            <InfoDetail label="Aktenummer" value={Akte.aktenummer} />
            <InfoDetail label="Registerjaar" value={Akte.registerjaar} />
            <InfoDetail
              el="div"
              label="Download PDF"
              value={
                <DocumentList
                  documents={Akte?.documents}
                  isExpandedView={false}
                />
              }
            />
          </>
        )}
      </PageContent>
    </DetailPage>
  );
}
