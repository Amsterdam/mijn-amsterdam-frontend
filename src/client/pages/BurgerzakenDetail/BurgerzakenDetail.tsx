import React from 'react';
import { useParams } from 'react-router-dom';
import { AppRoutes, ChapterTitles } from '../../../universal/config';
import {
  capitalizeFirstLetter,
  defaultDateFormat,
  isError,
  isLoading,
} from '../../../universal/helpers';
import {
  Alert,
  ChapterIcon,
  DetailPage,
  InfoDetail,
  LoadingContent,
  PageContent,
  PageHeading,
} from '../../components';
import styles from './BurgerzakenDetail.module.scss';
import { useAppStateGetter } from '../../hooks/useAppState';

export default function BurgerzakenDetail() {
  const { BRP } = useAppStateGetter();
  const { id } = useParams<{ id: string }>();

  const DocumentItem = BRP.content?.identiteitsbewijzen?.find(
    (item) => item.id === id
  );
  const noContent = !isLoading && !DocumentItem;

  return (
    <DetailPage>
      <PageHeading
        icon={<ChapterIcon />}
        backLink={{
          to: AppRoutes.BURGERZAKEN,
          title: ChapterTitles.BURGERZAKEN,
        }}
        isLoading={isLoading(BRP)}
      >
        {capitalizeFirstLetter(DocumentItem?.title || 'Document')}
      </PageHeading>

      <PageContent className={styles.DetailPageContent}>
        {(isError(BRP) || noContent) && (
          <Alert type="warning">
            <p>We kunnen op dit moment geen gegevens tonen.</p>
          </Alert>
        )}
        {isLoading(BRP) && (
          <LoadingContent className={styles.LoadingContentInfo} />
        )}
        {!!DocumentItem && (
          <>
            <InfoDetail
              label="Documentnummer"
              value={DocumentItem.documentNummer}
            />
            <InfoDetail
              label="Datum uitgifte"
              value={defaultDateFormat(DocumentItem.datumUitgifte)}
            />
            <InfoDetail
              label="Geldig tot"
              value={defaultDateFormat(DocumentItem.datumAfloop)}
            />
          </>
        )}
      </PageContent>
    </DetailPage>
  );
}
