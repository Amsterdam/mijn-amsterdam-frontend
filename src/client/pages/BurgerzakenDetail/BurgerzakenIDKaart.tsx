import { useParams } from 'react-router-dom';

import styles from './BurgerzakenDetail.module.scss';
import { AppRoutes } from '../../../universal/config/routes';
import { isError, isLoading } from '../../../universal/helpers/api';
import { defaultDateFormat } from '../../../universal/helpers/date';
import { capitalizeFirstLetter } from '../../../universal/helpers/text';
import {
  DetailPage,
  ErrorAlert,
  InfoDetail,
  LoadingContent,
  PageContent,
  PageHeading,
  ThemaIcon,
} from '../../components';
import { ThemaTitles } from '../../config/thema';
import { useAppStateGetter } from '../../hooks/useAppState';

export default function BurgerzakenIDKaart() {
  const { BRP } = useAppStateGetter();
  const { id } = useParams<{ id: string }>();

  const DocumentItem = BRP.content?.identiteitsbewijzen?.find(
    (item) => item.id === id
  );
  const noContent = !isLoading && !DocumentItem;

  return (
    <DetailPage>
      <PageHeading
        icon={<ThemaIcon />}
        backLink={{
          to: AppRoutes.BURGERZAKEN,
          title: ThemaTitles.BURGERZAKEN,
        }}
        isLoading={isLoading(BRP)}
      >
        {capitalizeFirstLetter(DocumentItem?.title || 'Document')}
      </PageHeading>

      <PageContent className={styles.DetailPageContent}>
        {(isError(BRP) || noContent) && (
          <ErrorAlert>We kunnen op dit moment geen gegevens tonen.</ErrorAlert>
        )}
        {isLoading(BRP) && (
          <LoadingContent className={styles.LoadingContentInfo} />
        )}
        {!!DocumentItem && (
          <>
            <InfoDetail
              label="Documentnummer"
              value={DocumentItem.documentNummer}
            />
            <InfoDetail
              label="Datum uitgifte"
              value={defaultDateFormat(DocumentItem.datumUitgifte)}
            />
            <InfoDetail
              label="Geldig tot"
              value={defaultDateFormat(DocumentItem.datumAfloop)}
            />
          </>
        )}
      </PageContent>
    </DetailPage>
  );
}
