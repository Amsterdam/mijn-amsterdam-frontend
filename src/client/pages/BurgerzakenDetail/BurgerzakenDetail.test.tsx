import { render } from '@testing-library/react';

import { generatePath } from 'react-router-dom';
import { MutableSnapshot } from 'recoil';
import { AppRoutes } from '../../../universal/config';
import { appStateAtom } from '../../hooks/useAppState';
import MockApp from '../MockApp';
import BurgerzakenDetail from './BurgerzakenDetail';

const identiteitsbewijzen = [
  {
    datumAfloop: '2025-10-15T00:00:00Z',
    datumUitgifte: '2015-10-15T00:00:00Z',
    documentNummer: 'PP57XKG54',
    documentType: 'paspoort',
    id: 'een-hash-van-documentnummer-1',
  },
  {
    datumAfloop: '2020-09-11T00:00:00Z',
    datumUitgifte: '2010-09-11T00:00:00Z',
    documentNummer: 'IE9962819',
    documentType: 'europese identiteitskaart',
    id: 'een-hash-van-documentnummer-2',
  },
];

const testState: any = {
  BRP: { status: 'OK', content: { identiteitsbewijzen } },
};

function initializeState(snapshot: MutableSnapshot) {
  snapshot.set(appStateAtom, testState);
}

describe('<BurgerzakenDetail />', () => {
  const routeEntry = generatePath(AppRoutes.BURGERZAKEN_DOCUMENT, {
    id: identiteitsbewijzen[0].id,
  });
  const routePath = AppRoutes.BURGERZAKEN_DOCUMENT;

  const Component = () => (
    <MockApp
      routeEntry={routeEntry}
      routePath={routePath}
      component={BurgerzakenDetail}
      initializeState={initializeState}
    />
  );

  it('Matches the Full Page snapshot', () => {
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });
});
