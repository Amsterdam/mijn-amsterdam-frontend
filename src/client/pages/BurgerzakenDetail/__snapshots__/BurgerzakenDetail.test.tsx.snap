// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<BurgerzakenDetail /> > Matches the Full Page snapshot 1`] = `
<DocumentFragment>
  <main
    class="_Page_47d7c0 _DetailPage_47d7c0"
  >
    <header
      class="_PageHeading_3b337d _hasIcon_3b337d _hasBackLink_3b337d"
    >
      <span
        aria-hidden="true"
        class="_Icon_3b337d"
      >
        <svg
          aria-label="Burgerzaken"
          class="_ChapterIcon_3ea2dd"
          fill="black"
          height="32px"
          viewBox="0 0 32 32"
          width="32px"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <g
            transform="translate(-5 0)"
          >
            <path
              d="M17.6,5 L27,5 L27,2.52631579 L17.6,5 Z M29,5 L32,5 L32,32 L10,32 L10,5 L29,0 L29,5 Z M26,25 L16,25 L16,22 C16,21 17,20 18,20 L24,20 C25,20 26,21 26,22 L26,25 Z M21,18 C19.3431458,18 18,16.6568542 18,15 C18,13.3431458 19.3431458,12 21,12 C22.6568542,12 24,13.3431458 24,15 C24,16.6568542 22.6568542,18 21,18 Z M14,27.96 L28,27.96 L28,9 L14,9 L14,27.96 Z"
            />
          </g>
        </svg>
      </span>
      <div
        class="_HeadingInner_3b337d"
      >
        <a
          class="_Button_c3ab45 _Button__plain_c3ab45 _Button__lean_c3ab45 _Linkd_c3ab45 _BackLink_3b337d"
          href="/burgerzaken"
        >
          <svg
            aria-hidden="true"
            class="_Icon_c3ab45 _Icon__left_c3ab45"
            fill="#000"
            focusable="false"
            height="14px"
            viewBox="0 0 32 32"
            width="14px"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <title>
              chevron-links-icoon
            </title>
            <polygon
              points="22.8571429 32 6.85714286 16 22.8571429 0 25.7568858 2.91002571 12.6771943 16 25.7568858 29.0899743"
            />
          </svg>
          Burgerzaken
        </a>
        <h2
          class="ams-heading ams-heading--1"
        >
          Document
        </h2>
      </div>
    </header>
    <div
      class="_PageContent_47d7c0 _DetailPageContent_4b8286"
    >
      <div
        class="_InfoDetail_6f1df2"
      >
        <h3
          class="ams-heading ams-heading--4 _Label_6f1df2"
        >
          Documentnummer
        </h3>
        <p
          class="_Value_6f1df2"
        >
          PP57XKG54
        </p>
      </div>
      <div
        class="_InfoDetail_6f1df2"
      >
        <h3
          class="ams-heading ams-heading--4 _Label_6f1df2"
        >
          Datum uitgifte
        </h3>
        <p
          class="_Value_6f1df2"
        >
          15 oktober 2015
        </p>
      </div>
      <div
        class="_InfoDetail_6f1df2"
      >
        <h3
          class="ams-heading ams-heading--4 _Label_6f1df2"
        >
          Geldig tot
        </h3>
        <p
          class="_Value_6f1df2"
        >
          15 oktober 2025
        </p>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
